src/OrleansProviders/Streams/Common/PooledCache/FixedSizeBuffer.cs

```diff
@@ -30,33 +29,14 @@ public FixedSizeBuffer(int blockSizeInByte)
         {
             if (blockSizeInByte < 0)
             {
-                throw new ArgumentOutOfRangeException("blockSize", "blockSize must be positive value.");
+                throw new ArgumentOutOfRangeException(nameof(blockSizeInByte), "blockSize must be positive value.");
             }
             count = 0;
             this.SizeInByte = blockSizeInByte;
             buffer = new byte[this.SizeInByte];
         }
 
         /// <summary>
-        /// Sets the purge callback that will be called when this buffer is being purged.  It notifies
-        ///   users of the buffer that the buffer is no longer valid.  This class is passed to the purge as a
-        ///   disposable.  When all resources referencing this buffer are released, this buffer needs be disposed.
-        /// </summary>
-        /// <param name="purge"></param>
-        public void SetPurgeAction(Action<IDisposable> purge)
```

Comment:
PurgeAction belongs to the old behavior where block pool trigger purging through calling PurgeAction. Now the only authority for purge is in eviction strategy, so remove this legacy.